### PR TITLE
Cut v2.23.0-rc.0

### PR DESCRIPTION
**Pull request by codesome**
~I am waiting for https://github.com/prometheus/prometheus/pull/6815 to be merged and~ have to upgrade the react dependancies before I cut the rc. Opening the PR early to finalise the changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Reduce allocated memory and latency of Hash method making queries a bit faster with less memory allocation. #8025
+* [ENHANCEMENT] Create a checkpoint only once at the end of all head compactions if there were multiple in a loop avoiding the expensive checkpointing loop. #8067
+* [ENHANCEMENT] Log when head compaction results in an empty block. #8037
+* [ENHANCEMENT] promtool tsdb list now prints block sizes. #7993
+* [BUGFIX] Fix button display when there is no panels. #8155
+* [BUGFIX] Fix timestamp() method for vector selector inside parenthesis. #8164
+* [BUGFIX] Fix panic with double close() of channel on calling `/-/quit/`. #8166
+* [BUGFIX] Fixed WAL corruption on partial writes within a page causing `invalid checksum` error on WAL replay. #8125
+* [BUGFIX] Update config metrics `prometheus_config_last_reload_successful` and `prometheus_config_last_reload_success_timestamp_seconds` right after initial validation before starting TSDB.
+* [BUGFIX] Correctly detect duplicate label names in exposition.
```
Comment (brian-brazil) [inline CHANGELOG.md:23]: This is specific to promtool, scraping already does this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+     - `retried_samples_total` -> `samples_retried_total`.
+     - `dropped_samples_total` -> `samples_dropped_total`.
+     - `pending_samples` -> `samples_pending`.
+* [FEATURE] Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `metadata_total` (total number of metadata sent to remote storage), `metadata_failed_total` (total number of metadata send failed, non recoverable errors), `metadata_retried_total` (total number of metadata send retried after facing recoverable errors), `metadata_bytes_total` (total bytes send for metadata after compression). #6815
+* [ENHANCEMENT] Add a metric `max_samples_per_send` for tracking the `max_samples_per_send` config of remote write. #8102
+* [ENHANCEMENT] Make the snapshot directory name always the same length. #8138
+* [ENHANCEMENT] Reduce allocated memory and latency of Hash method making queries a bit faster with less memory allocation. #8025
+* [ENHANCEMENT] Create a checkpoint only once at the end of all head compactions if there were multiple in a loop avoiding the expensive checkpointing loop. #8067
+* [ENHANCEMENT] Log when head compaction results in an empty block. #8037
```
Comment (brian-brazil) [inline CHANGELOG.md:16]: We don't usually mention changes to logging.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+
+* [CHANGE] Make the React UI default. #8142
+* [CHANGE] The following metrics were removed/renamed in remote write. #6815
+     - `succeeded_samples_total` was removed and `samples_total` was introduced for all the samples attempted to send.
+     - `sent_bytes_total` was removed and was replaced with `samples_bytes_total` and `metadata_bytes_total`.
+     - `failed_samples_total` -> `samples_failed_total` .
+     - `retried_samples_total` -> `samples_retried_total`.
+     - `dropped_samples_total` -> `samples_dropped_total`.
+     - `pending_samples` -> `samples_pending`.
+* [FEATURE] Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `metadata_total` (total number of metadata sent to remote storage), `metadata_failed_total` (total number of metadata send failed, non recoverable errors), `metadata_retried_total` (total number of metadata send retried after facing recoverable errors), `metadata_bytes_total` (total bytes send for metadata after compression). #6815
```
Comment (brian-brazil) [inline CHANGELOG.md:11]: Only the metric names need to be mentioned, this isn't the place for metric docs.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Add a metric `max_samples_per_send` for tracking the `max_samples_per_send` config of remote write. #8102
```
Comment (brian-brazil) [inline CHANGELOG.md:12]: Same here, we only need the metric name.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Make the snapshot directory name always the same length. #8138
+* [ENHANCEMENT] Reduce allocated memory and latency of Hash method making queries a bit faster with less memory allocation. #8025
```
Comment (brian-brazil) [inline CHANGELOG.md:14]: I'd not mention memory allocation, as that's just CPU here. Just "Improved performance of Hash method" would do.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Create a checkpoint only once at the end of all head compactions if there were multiple in a loop avoiding the expensive checkpointing loop. #8067
```
Comment (brian-brazil) [inline CHANGELOG.md:15]: This is overly lengthy, if someone wants details they can check the PR. You can end this at compactions.

**Review by brian-brazil — COMMENTED** [review]
8074 is an enhancement, 8177 is an bugfix, 8050 is an enhancement, 8060 is a change/enhancement, 8096 is an enhancement

Should 8051 be mentioned?

File: CHANGELOG.md (hunk 3 above)
Comment (roidelapluie) [inline CHANGELOG.md:11]: `[EXPERIMENTAL]`

**Comment by roidelapluie** [comment]
I note that you do not add categories in changelog like we use to

**Comment by roidelapluie** [comment]
You miss #8074

**Comment by codesome** [comment]
> Should 8051 be mentioned?

I guess not. 8051 just adds additional information to the error log.

**Comment by codesome** [comment]
Ready for another review!

**Comment by codesome** [comment]
I have included https://github.com/prometheus/prometheus/pull/8192 too since I synced the release branch with master now

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,32 @@
...
+* [CHANGE] UI: Make the React UI default. #8142
+* [CHANGE] Remote write: The following metrics were removed/renamed in remote write. #6815
+     - `succeeded_samples_total` was removed and `samples_total` was introduced for all the samples attempted to send.
+     - `sent_bytes_total` was removed and was replaced with `samples_bytes_total` and `metadata_bytes_total`.
+     - `failed_samples_total` -> `samples_failed_total` .
+     - `retried_samples_total` -> `samples_retried_total`.
+     - `dropped_samples_total` -> `samples_dropped_total`.
+     - `pending_samples` -> `samples_pending`.
+* [CHANGE][ENHANCEMENT] Remote: Do not collect non-initialized timestamp metrics. #8060
+* [FEATURE][EXPERIMENTAL] Remote write: Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `metadata_total`, `metadata_failed_total`, `metadata_retried_total`, `metadata_bytes_total`. #6815
```
Comment (roidelapluie) [inline CHANGELOG.md:12]: should we include the namespace of the metrics?

Comment (roidelapluie) [reply CHANGELOG.md:12]: or not mention the metrics

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+* [ENHANCEMENT] Remote write: Added a metric `max_samples_per_send` for remote write. #8102
```
Comment (roidelapluie) [inline CHANGELOG.md:13]: should we include the namespace of the metrics?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,32 @@
...
+* [ENHANCEMENT] TSDB: Make the snapshot directory name always the same length. #8138
+* [ENHANCEMENT] TSDB: Create a checkpoint only once at the end of all head compactions. #8067
+* [ENHANCEMENT] TSDB: Avoid Series API from hitting the chunks. #8050 
+* [ENHANCEMENT] TSDB: Cache label name and last value when adding series during compactions making compactions faster. #8192
+* [ENHANCEMENT] PromQL: Improved performance of Hash method making queries a bit faster. #8025
```
Comment (roidelapluie) [inline CHANGELOG.md:18]: is it worth mentioning?

Comment (codesome) [reply CHANGELOG.md:18]: I think so yes

File: CHANGELOG.md (hunk 7 above)
Comment (codesome) [reply CHANGELOG.md:12]: Yes makes sense if you mean prefix them with `prometheus_remote_storage_`

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+
+* [CHANGE] UI: Make the React UI default. #8142
+* [CHANGE] Remote write: The following metrics were removed/renamed in remote write. #6815
+     - `prometheus_remote_storage_succeeded_samples_total` was removed and `prometheus_remote_storage_samples_total` was introduced for all the samples attempted to send.
+     - `prometheus_remote_storage_sent_bytes_total` was removed and was replaced with `prometheus_remote_storage_samples_bytes_total` and `prometheus_remote_storage_metadata_bytes_total`.
+     - `prometheus_remote_storage_failed_samples_total` -> `prometheus_remote_storage_samples_failed_total` .
+     - `prometheus_remote_storage_retried_samples_total` -> `prometheus_remote_storage_samples_retried_total`.
+     - `prometheus_remote_storage_dropped_samples_total` -> `prometheus_remote_storage_samples_dropped_total`.
+     - `prometheus_remote_storage_pending_samples` -> `prometheus_remote_storage_samples_pending`.
+* [CHANGE][ENHANCEMENT] Remote: Do not collect non-initialized timestamp metrics. #8060
```
Comment (roidelapluie) [inline CHANGELOG.md:11]:
```suggestion
* [CHANGE] [ENHANCEMENT] Remote: Do not collect non-initialized timestamp metrics. #8060
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,32 @@
...
+* [FEATURE][EXPERIMENTAL] Remote write: Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `prometheus_remote_storage_metadata_total`, `prometheus_remote_storage_metadata_failed_total`, `prometheus_remote_storage_metadata_retried_total`, `prometheus_remote_storage_metadata_bytes_total`. #6815
```
Comment (roidelapluie) [inline CHANGELOG.md:12]:
```suggestion
* [FEATURE] [EXPERIMENTAL] Remote write: Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `prometheus_remote_storage_metadata_total`, `prometheus_remote_storage_metadata_failed_total`, `prometheus_remote_storage_metadata_retried_total`, `prometheus_remote_storage_metadata_bytes_total`. #6815
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,32 @@
+## 2.23.0-rc.0 / 2020-11-20
+
+* [CHANGE] UI: Make the React UI default. #8142
+* [CHANGE] Remote write: The following metrics were removed/renamed in remote write. #6815
+     - `prometheus_remote_storage_succeeded_samples_total` was removed and `prometheus_remote_storage_samples_total` was introduced for all the samples attempted to send.
+     - `prometheus_remote_storage_sent_bytes_total` was removed and was replaced with `prometheus_remote_storage_samples_bytes_total` and `prometheus_remote_storage_metadata_bytes_total`.
```
Comment (roidelapluie) [inline CHANGELOG.md:6]:
```suggestion
     - `prometheus_remote_storage_sent_bytes_total` was removed and replaced with `prometheus_remote_storage_samples_bytes_total` and `prometheus_remote_storage_metadata_bytes_total`.
```

**Comment by brian-brazil** [comment]
:+1:

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,32 @@
...
+     - `prometheus_remote_storage_pending_samples` -> `prometheus_remote_storage_samples_pending`.
+* [CHANGE] Remote: Do not collect non-initialized timestamp metrics. #8060
+* [FEATURE] [EXPERIMENTAL] Remote write: Allow metric metadata to be propagated via remote write. The following new metrics were introduced: `prometheus_remote_storage_metadata_total`, `prometheus_remote_storage_metadata_failed_total`, `prometheus_remote_storage_metadata_retried_total`, `prometheus_remote_storage_metadata_bytes_total`. #6815
+* [ENHANCEMENT] Remote write: Added a metric `prometheus_remote_storage_max_samples_per_send` for remote write. #8102
+* [ENHANCEMENT] TSDB: Make the snapshot directory name always the same length. #8138
+* [ENHANCEMENT] TSDB: Create a checkpoint only once at the end of all head compactions. #8067
+* [ENHANCEMENT] TSDB: Avoid Series API from hitting the chunks. #8050 
+* [ENHANCEMENT] TSDB: Cache label name and last value when adding series during compactions making compactions faster. #8192
+* [ENHANCEMENT] PromQL: Improved performance of Hash method making queries a bit faster. #8025
+* [ENHANCEMENT] promtool: tsdb list now prints block sizes. #7993
```
Comment (roidelapluie) [inline CHANGELOG.md:19]:
```suggestion
* [ENHANCEMENT] promtool: `tsdb list` now prints block sizes. #7993
```

**Review by roidelapluie — COMMENTED** [review]
:+1: one last readability nit if you feel up to :)

File: CHANGELOG.md (hunk 13 above)
Comment (codesome) [reply CHANGELOG.md]: Fixed it :) also changed the date since I plan to cut it on Monday

**Comment by roidelapluie** [comment]
Do not forget to change the VERSION file

**Review by roidelapluie — CHANGES_REQUESTED** [review]
VERSION file should be changed too

**Comment by roidelapluie** [comment]
Compactions seem to be faster, not sure how to explain that.

**Comment by codesome** [comment]
> VERSION file should be changed too

:facepalm: have to wait for another build then

**Comment by codesome** [comment]
Compaction time improvements look awesome! The memory RSS is also less, I wonder why. If the prombench is hitting the series API (is it @geekodour?), then `[ENHANCEMENT] TSDB: Avoid Series API from hitting the chunks. #8050` could explain something, but that should still not give us 33 GB vs 35.5 GB I guess.

**Comment by geekodour** [comment]
> If the prombench is hitting the series API
The querier does seem to use it: https://github.com/prometheus/test-infra/blob/master/tools/load-generator/main.py#L47

Also, the github-prombench comment-monitor seems to be down atm, checking. cancelling the test manually as of now.

**Comment by codesome** [comment]
Thanks @geekodour!

**Comment by codesome** [comment]
And looking at load generator, it is actually not hitting the /series API but just /query and /query_range. So that enhancement is ruled out. Anyway, not a blocker since it's a reduction in RSS :)

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**